src/types.js

```diff
@@ -27,10 +27,28 @@
  * @property { (arg: WorkerArg) => void } [onTerminateWorker] A callback that is called whenever a worker is being terminated. It can be used to release resources that might have been allocated for this specific worker. The callback is passed as argument an object as described for `onCreateWorker`, with each property sets with the value for the worker being terminated.
  */
 
+
+/**
+  * @typedef {Object} AbortStartArgs
+  * @property {number} [id] identifier of the task which is starting its abort operation.
+  * @property {PromiseLike<void>} [abortPromise] PromiseLike Object which resolves or rejects when the abort operation concludes.
+  *
+  */
+
+/**
+  * @typedef {Object} AbortResolutionArgs
+  * @property {Error | undefined} [error] An error which occured during the abort operation. If an error did not occure the value will be `undefined`.
+  * @property {number} [id] identifier of the task.
+  * @property {boolean} [isTerminating] A flag which indicates the termination status of the worker which ececuted the task.
+  */
+
 /**
  * @typedef {Object} ExecOptions
  * @property {(payload: any) => unknown} [on] An event listener, to handle events sent by the worker for this execution.
  * @property {Object[]} [transfer] A list of transferable objects to send to the worker. Not supported by `process` worker type. See ./examples/transferableObjects.js for usage.
+ * @property {(payload: any) => AbortResolutionArgs} [onAbortResolution] An event listener triggered when whenever an abort operation concludes.
```

Comment:
Is there an actual use case for the `onAbortResolution` and `onAbortStart` callbacks? If there is no clear use case I would prefer to not implement it right now, to keep things simple.

Reply:
The main usage is to give more context to the user on life cycle of the task. but `onAbortStart` does provide access the to promise.
If we are to remove these callbacks then we should keep `abortResolver` so the user knows when the abort operation concludes.